File: app/src/debug/java/com/stedi/randomimagegenerator/app/di/modules/BuildTypeDependentModule.java
package com.stedi.randomimagegenerator.app.di.modules;

import android.content.Context;

import com.stedi.randomimagegenerator.app.model.repository.CachedPresetRepository;
import com.stedi.randomimagegenerator.app.model.repository.DatabasePresetRepository;
import com.stedi.randomimagegenerator.app.model.repository.PresetRepository;
import com.stedi.randomimagegenerator.app.other.logger.LogCatLogger;
import com.stedi.randomimagegenerator.app.other.logger.Logger;

import javax.inject.Named;
import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class BuildTypeDependentModule {
    @Provides
    @Singleton
    Logger provideLogger() {
        return new LogCatLogger("RIGAPP-DBG");
    }

    @Provides
    @Singleton
    PresetRepository providePresetRepository(@Named("AppContext") Context context, Logger logger) {
        return new CachedPresetRepository(new DatabasePresetRepository(context, logger));
    }
}

File: app/src/debug/java/com/stedi/randomimagegenerator/app/di/modules/BuildTypeDependentModule.kt
package com.stedi.randomimagegenerator.app.di.modules

import android.content.Context
import com.stedi.randomimagegenerator.app.di.AppContext
import com.stedi.randomimagegenerator.app.model.repository.CachedPresetRepository
import com.stedi.randomimagegenerator.app.model.repository.DatabasePresetRepository
import com.stedi.randomimagegenerator.app.model.repository.PresetRepository
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class BuildTypeDependentModule {

    @Provides
    @Singleton
    fun providePresetRepository(@AppContext context: Context): PresetRepository {
        return CachedPresetRepository(DatabasePresetRepository(context))
    }
}